File: spec/python/test_repeat_n_strz.py
import unittest

from repeat_n_strz import RepeatNStrz

class TestRepeatNStrz(unittest.TestCase):
    def test_repeat_n_strz(self):
        with RepeatNStrz.from_file("src/repeat_n_strz.bin") as r:

            self.assertEqual(r.qty, 2)
            self.assertEqual(r.lines, ["foo", "bar"])

File: spec/cpp_stl/test_repeat_n_strz.cpp
#include <boost/test/unit_test.hpp>

#include <repeat_n_strz.h>

#include <iostream>
#include <fstream>
#include <vector>

BOOST_AUTO_TEST_CASE(test_repeat_n_strz) {
    std::ifstream ifs("src/repeat_n_strz.bin", std::ifstream::binary);
    kaitai::kstream ks(&ifs);
    repeat_n_strz_t* r = new repeat_n_strz_t(&ks);

    BOOST_CHECK_EQUAL(r->qty(), 2);

    std::vector<std::string> exp_vector;
    exp_vector.push_back(std::string("foo"));
    exp_vector.push_back(std::string("bar"));
    
    BOOST_CHECK_EQUAL_COLLECTIONS(
        r->lines()->begin(), r->lines()->end(),
        exp_vector.begin(), exp_vector.end()
    );
}
